misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java

```diff
@@ -0,0 +1,408 @@
+     */
+    public static ArrayList<TypeDefinitionNode> fromJSON(String jsonString) throws ConverterException, IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        JsonNode inputJsonObject = objectMapper.readTree(jsonString);
+        Map<String, Object> schema = SchemaGenerator.generate(inputJsonObject);
+        String schemaJson = objectMapper.writeValueAsString(schema);
+        OpenAPI model = parseJSONSchema(schemaJson);
+        return generateRecords(model);
+    }
+
+    /**
+     * Generates Ballerina Record nodes for given OpenAPI model.
+     *
+     * @param openApi OpenAPI model
+     * @return {@link ArrayList<TypeDefinitionNode>}  List of Record Nodes
+     * @throws ConverterException in case of bad record fields
+     */
+    public static ArrayList<TypeDefinitionNode> generateRecords(OpenAPI openApi) throws ConverterException {
+        // TypeDefinitionNodes their
+        typeDefinitionNodeList = new LinkedList<>();
+
+        if (openApi.getComponents() != null) {
+            //Create typeDefinitionNode
+            Components components = openApi.getComponents();
+            if (components.getSchemas() != null) {
+                Map<String, Schema> schemas = components.getSchemas();
+                for (Map.Entry<String, Schema> schema: schemas.entrySet()) {
+                    List<String> required = schema.getValue().getRequired();
+
+                    //1.typeKeyWord
+                    Token typeKeyWord = AbstractNodeFactory.createIdentifierToken("type");
+                    //2.typeName
+                    IdentifierToken typeName = AbstractNodeFactory.createIdentifierToken(
+                            escapeIdentifier(schema.getKey().trim()));
+                    //3.typeDescriptor - RecordTypeDescriptor
+                    //3.1 recordKeyWord
+                    Token recordKeyWord = AbstractNodeFactory.createIdentifierToken("record");
+                    //3.2 bodyStartDelimiter
+                    Token bodyStartDelimiter = AbstractNodeFactory.createIdentifierToken("{");
+                    //3.3 fields
+                    //Generate RecordFiled
+                    List<Node> recordFieldList = new ArrayList<>();
+                    Schema schemaValue = schema.getValue();
+                    if (schema.getValue().getProperties() != null || schema.getValue() instanceof ObjectSchema) {
+                        Map<String, Schema> fields = schema.getValue().getProperties();
+                        if (fields != null) {
+                            for (Map.Entry<String, Schema> field : fields.entrySet()) {
+                                addRecordFields(required, recordFieldList, field);
+                            }
+                        }
+                        NodeList<Node> fieldNodes = AbstractNodeFactory.createNodeList(recordFieldList);
+                        Token bodyEndDelimiter = AbstractNodeFactory.createIdentifierToken("}");
```

Comment:
If possible shall we create Token using SyntaxKind instead of using string `Token bodyEndDelimiter = AbstractNodeFactory.createToken(OPEN_BRACE_TOKEN);` please check other token as well :)

Reply:
Addressed with 973d779